src/core/public/chrome/ui/header/header_search_bar.tsx

```diff
@@ -0,0 +1,240 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import {
+  EuiButtonIcon,
+  EuiFieldSearch,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiListGroup,
+  EuiListGroupItem,
+  EuiPanel,
+  EuiPopover,
+  EuiResizeObserver,
+  EuiText,
+  EuiTitle,
+  EuiToolTip,
+} from '@elastic/eui';
+import React, { ReactNode, useCallback, useRef, useState } from 'react';
+import { i18n } from '@osd/i18n';
+import { GlobalSearchStrategy, SearchObjectTypes } from '../../global_search';
+
+interface Props {
+  globalSearchStrategies: GlobalSearchStrategy[];
+  panel?: boolean;
+  onSearchResultClick?: () => void;
+}
+
+export const HeaderSearchBarIcon = ({ globalSearchStrategies }: Props) => {
+  const [isSearchPopoverOpen, setIsSearchPopoverOpen] = useState(false);
+  const buttonRef = useRef<HTMLButtonElement>(null);
+  return (
+    <EuiPopover
+      panelPaddingSize="s"
+      anchorPosition="downCenter"
+      repositionOnScroll={true}
+      isOpen={isSearchPopoverOpen}
+      closePopover={() => {
+        setIsSearchPopoverOpen(false);
+      }}
+      button={
+        <EuiToolTip
+          content={i18n.translate('core.globalSearch.icon.toolTip', {
+            defaultMessage: 'Search',
+          })}
+        >
+          <EuiButtonIcon
+            aria-label="search"
+            iconType="search"
+            color="text"
+            buttonRef={buttonRef}
+            data-test-subj="search-icon"
+            onClick={() => {
+              setIsSearchPopoverOpen(!isSearchPopoverOpen);
+              buttonRef.current?.blur();
+            }}
+          />
+        </EuiToolTip>
+      }
+    >
+      <EuiPanel
+        hasBorder={false}
+        hasShadow={false}
+        paddingSize="none"
+        style={{ minHeight: '300px', minWidth: '400px' }}
+      >
+        <HeaderSearchBar
+          globalSearchStrategies={globalSearchStrategies}
+          panel
+          onSearchResultClick={() => {
+            setIsSearchPopoverOpen(false);
+            buttonRef.current?.blur();
+          }}
+        />
+      </EuiPanel>
+    </EuiPopover>
+  );
+};
+
+export const HeaderSearchBar = ({ globalSearchStrategies, panel, onSearchResultClick }: Props) => {
+  const [results, setResults] = useState([] as React.JSX.Element[]);
+  const [isLoading, setIsLoading] = useState(false);
+  const [isPopoverOpen, setIsPopoverOpen] = useState(false);
+  const [panelWidth, setPanelWidth] = useState(0);
+  const inputRef = (node: HTMLElement | null) => setInputEl(node);
+  const [inputEl, setInputEl] = useState<HTMLElement | null>(null);
```

Comment:
This is by intention, inputEl is used as dependency for resize.

```ts
const onResize = useCallback(() => {
    if (inputEl) {
      const width = inputEl.getBoundingClientRect().width;
      setPanelWidth(width);
    }
  }, [inputEl, setPanelWidth]);
```

Reply:
Found that we can remove `onResize` as it will adjust the panel width when focus.

and changed to `const inputElRef = useRef<HTMLElement | null>(null);` as well.